app/src/main/java/com/example/pocketcloset/Fragments/ProfileFragment.java

```diff
@@ -121,4 +274,175 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
+            Collections.sort(clothing, (d1, d2) -> {
+                return d2.getWornCount() - d1.getWornCount();
+            });
+            Collections.reverse(clothing);
+        }
+
+        if(clothing != null && clothing.size() > 0) {
+            return clothing.get(0);
+        }
+        return null;
+    }
+
+
+
+
+    protected void queryMostWornClothes() {
+        // specify what type of data we want to query - Post.class
+        ParseQuery<Clothing> query = ParseQuery.getQuery(Clothing.class);
+        // include data referred by user key
+        Log.i("WardrobeFragment", "queryClothes(with params) ran");
+        query.include(Clothing.KEY_CLOTHING_IMAGE);
+        query.whereEqualTo(Clothing.KEY_USER, user);
+
+        // limit query to latest 20 items
+        query.setLimit(20);
+        // order posts by creation date (newest first)
+        query.addDescendingOrder("createdAt");
+        // start an asynchronous call for posts
+        query.findInBackground(new FindCallback<Clothing>() {
+            @Override
+            public void done(List<Clothing> clothes, ParseException e) {
+                // check for errors
+                if (e != null) {
+                    Log.e("ProfileFragment", "Issue with getting clothing", e);
+                    return;
+                }
+                // save received posts to list and notify adapter of new data'
+                mostWornAdapter.clear();
+                allClothing.addAll(clothes);
+                mostWornAdapter.notifyDataSetChanged();
+
+                if((MostWorn(ClothingType.TOP) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.TOP));
+                }
+                if((MostWorn(ClothingType.HEADWEAR) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.HEADWEAR));
+                }
+                if((MostWorn(ClothingType.PANTS) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.PANTS));
+                }
+                if((MostWorn(ClothingType.DRESS) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.DRESS));
+                }
+                if((MostWorn(ClothingType.SHOES) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.SHOES));
+                }
+                if((MostWorn(ClothingType.OVERWEAR) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.OVERWEAR));
+                }
+                if((MostWorn(ClothingType.HANDHELD) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.HANDHELD));
+                }
+                if((MostWorn(ClothingType.BRACELET) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.BRACELET));
+                }
+                if((MostWorn(ClothingType.EARRINGS) != null)) {
+                    mostWorn.add(MostWorn(ClothingType.EARRINGS));
+                }
+
+
+
+            }
+        });
+
+
+    }
+
+    protected void queryLeastWornClothes() {
```

Comment:
This is very similar to the most worn function. Can these function pairs be refactored to share the majority of the logic?